web/massastation/src/custom/api/usePost.ts

```diff
@@ -1,34 +1,38 @@
-// STYLES
-
-// EXTERNALS
-import axios from 'axios';
+import axios, { AxiosResponse } from 'axios';
 import { UseMutationResult, useMutation } from '@tanstack/react-query';
 
-// LOCALS
-
-export function usePost<T, P = unknown>(
+/**
+ * @param resource - path of the resource
+ * @typeParam TBody - type of the request
+ * @typeParam TResponse - type of the response
+ * @returns
+ */
+export function usePost<TBody, TResponse = null>(
   resource: string,
 ): UseMutationResult<
-  P,
+  TResponse,
   unknown,
-  { params?: Record<string, string>; payload?: T },
+  { params?: Record<string, string>; payload?: TBody },
```

Comment:
I am not against those changes. But we'll end up with a format for usePost/usePut here that is not the same as in the wallet. It could be tricky. 
It's a non blocking but we MUST take care of that otherwise it will be messy. @Pierregvx pinging you here to don't forget too.

Reply:
I don't really agree, I don't think it's a problem to not have a standard, I don't see it as messy like you said. We can talk about this after the release.